ecommerce/extensions/payment/processors/payu.py

```diff
@@ -0,0 +1,238 @@
+        # VD
+        fechaActual = time.strftime("%H%M%S")
+        codigoReferencia = str(basket.order_number) + str(fechaActual)
+        # --
+        parameters = {
+            'merchantId': self.merchant_id,
+            'accountId': self.account_id,
+            'description': u'Inscripción {}'.format(course_id),
+            'referenceCode': codigoReferencia,
+            'amount': str(basket.total_incl_tax),
+            'tax': '0',
+            'taxReturnBase': '0',
+            'currency': basket.currency,
+            'signature': self._generate_signature(self.api_key, self.merchant_id, codigoReferencia, str(basket.total_incl_tax), str(basket.currency)),
+            'buyerEmail': user.email,
+            'responseUrl': responseUrl,
+            'confirmationUrl': confirmationUrl
+        }
+        parameters['payment_page_url'] = self.payment_page_url
+
+        if self.test:
+            parameters['test'] = self.test
+
+        return parameters
+
+    def handle_processor_response(self, response, basket=None):
+        available_attempts = 1
+
+        transaction_id = response['transactionId']
+        transactionState = response['transactionState']
+        status = response['lapTransactionState'] #REJECTED/PENDING/APPROVED DECLINED, ERROR, EXPIRED
+
+        self.record_processor_response(response, transaction_id=transaction_id, basket=basket)
+        logger.info(u"Successfully executed PayU payment [%s] for basket [%d].", transaction_id, basket.id)
+
+        if transactionState != '4':
+            exception = {
+                'cancel': UserCancelled,
+                'decline': TransactionDeclined,
+                'error': GatewayError
+            }.get(status, InvalidPayUStatus)
+
+            raise exception
+
+        currency = response['currency']
+        total = Decimal(response['TX_VALUE'])
+        email = basket.owner.email
+        label = 'PayU ({})'.format(email) if email else 'PayU Account'
+
+        return HandledProcessorResponse(
+            transaction_id=transaction_id,
+            total=total,
+            currency=currency,
+            card_number=label,
+            card_type=None
+        )
+
+    def issue_credit(self, source, amount, currency):
+        """
+        This method should be implemented in the future in order
+        to accept payment refunds
+        see http://developers.payulatam.com/en/api/refunds.html
+        """
+
+        logger.exception(
+            'PayU processor can not issue credits or refunds',
+        )
+
+        raise NotImplementedError
+
+    def _generate_signature(self, ApiKey, merchantId, referenceCode, amount, currency):
+        firma = hashlib.md5()
+        firma.update(ApiKey + "~" + merchantId + "~" + referenceCode + "~" + amount + "~" + currency)
+        wfirma = firma.hexdigest()
+
+        return wfirma
+
+    def _verify_student(self, site, username):
+        path_api = "/camrom/api/v1/change_to_verified_mode/"
```

Comment:
What about having this path_api as a configuration?

Reply:
yes, that'd be better. I'll make the changes.